Zeal/melody.cpp

```
@@ -93,10 +95,22 @@ void Melody::end()
         songs.clear();
         retry_count = 0;
         casting_melody_spell_id = kInvalidSpellId;
+        use_item_index = -1;
         Zeal::EqGame::print_chat(USERCOLOR_SPELL_FAILURE, "Your melody has ended.");
     }
 }
 
+bool Melody::use_item(int item_index)
+{
+    if (songs.empty())
```

Possibly change to i`f (songs.empty() || (item_index < 0))` to avoid silent failures (ignoring it).